src/AutorestSwift/Utils/String+Extensions.swift

```diff
@@ -89,13 +89,30 @@ extension String {
 
     var isReserved: Bool {
         let reserved = [
-            "Error",
-            "Int",
-            "String",
-            "Bool",
-            "Enum",
-            "Array"
+            "Error": true,
```

Comment:
What is the purpose of the flag ? And when it will be set to false?

Reply:
It would never be set to false. The whole purpose is to use efficient constant time hash lookups instead of O(n) lookups that you would have to do with arrays.